tests/security/ima/evm_protection_hmacs.pm

```diff
@@ -50,7 +50,7 @@ sub run {
     assert_script_run "setfattr -x security.ima $sample_app";
     validate_script_output "getfattr -m security.evm -d $sample_app", sub { m/^$/ };
 
-    replace_grub_cmdline_settings('evm=fix ima_appraise=fix', '', 1);
+    replace_grub_cmdline_settings('evm=fix ima_appraise=fix', '', update_grub => 1);
```

Comment:
BTW I wonder if it work as 
`replace_grub_cmdline_settings('evm=fix ima_appraise=fix', update_grub => 1);`
i.e. having `$new` as `undef`. It'd probably introduce a warning, there would need to be added support for it in `change_grub_config()`. And explicitly set new as `''` is more readable.

Reply:
`
  1029 sub change_grub_config {
  1030     die((caller(0))[3] . ' expects from 3 to 5 arguments') unless (@_ >= 3 && @_ <= 5);
  1031     my ($old, $new, $search, $modifiers, $update_grub) = @_;
  1032     $modifiers   //= '';
  1033     $update_grub //= 0;
+ 1034     $new //= '';
  1035     $search = "/$search/" if defined $search;
  1036     
  1037     assert_script_run("sed -ie '${search}s/${old}/${new}/${modifiers}' " . GRUB_DEFAULT_FILE);
`

Reply:
Yes, that's exactly what I meant (+ mention it in perldoc and commit message).

Reply:
http://mitigations.qa2.suse.asia/tests/5372#step/sestatus/37

@jouyingbin Please take a look about the grub operation.